gateway-js/src/datasources/RemoteGraphQLDataSource.ts

```diff
@@ -152,7 +153,7 @@ export class RemoteGraphQLDataSource<TContext extends Record<string, any> = Reco
       // Use the fetcher's `Request` implementation for compatibility
       fetchResponse = await this.fetcher(http.url, {
         ...http,
-        body: JSON.stringify(requestWithoutHttp)
+        body: requestBody
```

Comment:
This wasn't introduced by this PR, but the adjustments here bring it to mind again since we have the opportunity / it's tempting to pass the entire object (`{ ...http, body: ... }`) to this invocation of what is meant to be a [Fetch API-standard](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) `fetcher`, rather than creating it separately.

Put another way, it wasn't immediately clear to me why this `fetcher` can no longer accept the `RequestInit` type that we have created (which is what `fetchRequest` is) above.

That recalls the note in https://github.com/apollographql/federation/pull/188 (which made the change):

> I debugged a lot and found the ultimate cause - which is we need the Request to be constructed by the fetcher's own Request class instead of node fetch's Request class as done currently)

So that's me _not asking for a change_, but just noting that I found it more peculiar that `make-fetch-happen` _couldn't_ accept `RequestInit` and that we had to dance around that in this way.  Almost seems like a bug/limitation of `make-fetch-happen`.